pandas/tools/tests/test_util.py

```diff
@@ -57,6 +57,15 @@ def test_invalid_input(self):
         msg = "Input must be a list-like of list-likes"
         for X in invalid_inputs:
             tm.assertRaisesRegexp(TypeError, msg, cartesian_product, X=X)
+            
+    def test_large_input(self):
+        # test failure of large inputs on windows OS
+        X = np.arange(65536)
+        Y = np.arange(65535)
+        result1, result2 = cartesian_product([X, Y])
+        expected_size = X.size * Y.size
```

Comment:
I couldn't think of another way to cause the error without adding checking code to the `cartesian_product` function itself, e.g.

```python
if np.any(cumprodX < 0):
    raise RuntimeError(...)
```